spinalcordtoolbox/deepseg_/gm.py

```diff
@@ -182,7 +182,7 @@ def segment_volume(ninput_volume, model_name,
     model_abs_path = gmseg_model_challenge.get_file_path(model_path)
 
     # Padding/cropping data to match the input dimensions of the model
-    volume_data = ninput_volume.get_data()
+    volume_data = np.asanyarray(ninput_volume.dataobj)
```

Comment:
I'm torn on the replacement using `np.asanyarray(im.dataobj)`, because in theory I agree with the deprecation message that using `get_fdata` to get a consistent, predictable data type would be better. But it's true that this way replicates the old behaviour more closely, so it's less risky.

In practice, I guess this refactoring is fine because it's safe, and we can always grep the source code later for uses of `dataobj` and audit them more deeply.